src/agent.py

```diff
@@ -215,3 +241,30 @@ def _on_agent_relation_departed(self, event: ops.RelationDepartedEvent) -> None:
             self.charm.unit.status = ops.ActiveStatus(f"Failed to remove {agent_name}")
             return
         self.charm.unit.status = ops.ActiveStatus()
+
+    def _on_ready(self, event: IngressPerAppReadyEvent) -> None:
```

Comment:
I'm wondering if ingress related definitions can be prefixed with `ingress` to help avoid confusion to what is ready :)

Reply:
makes sense, I'm thinking of naming it `_ingress_on_ready` rather than `_on_ingress_ready` to stick with the naming convention